File: C++-Enum_templates_e_mais/Banco-4/Funcionario.cpp
#include "Funcionario.hpp"

Funcionario::Funcionario(Cpf cpf, std::string nome, float salario, diaDaSemana payDay) :
	Pessoa(cpf, nome), salario(salario), payDay(payDay)
{

}

std::string Funcionario::getNome() {
	return nome;
}

float Funcionario::getSalario() const {
	return salario;
}

File: BancoSharp/BancoSharp/Funcionarios/Funcionario.cs
namespace BancoSharp.Funcionarios
{
    public abstract class Funcionario
    {
        public static int TotalFuncionarios { get; private set; }
        public Funcionario(string nome, string cpf, double salario)
        {
            Nome = nome;
            Cpf = cpf;
            Salario = salario;
            TotalFuncionarios++;
        }
        public string Nome { get; protected set; }
        public string Cpf { get; protected set; }
        public double Salario { get; protected set; }

        public abstract double GetBonificacao();
        protected abstract void AumentaSalario();
        public void mostraInfosFuncionarios()
        {
            Console.WriteLine(" Nome do Funcionário: " + Nome);
            Console.WriteLine(" CPF: " + Cpf);
            Console.WriteLine(" Salário: " + Salario);
        }

    }
}
